### PR TITLE
pubsub: factor out common URL code

### PR DESCRIPTION
**Pull request by jba**
Fixes partially #1476.

### CODE REVIEW

File: pubsub/pubsub.go [hunk 1]
```diff
@@ -565,85 +565,71 @@ type SubscriptionURLOpener interface {
 //
 // The zero value is a multiplexer with no registered schemes.
 type URLMux struct {
-	subscriptionSchemes map[string]SubscriptionURLOpener
-	topicSchemes        map[string]TopicURLOpener
+	subscriptionSchemes urlSchemeMap
```
Comment (vangent) [inline pubsub/pubsub.go]: The "zero value" comment above is no longer true IIUC. It's a nice property. Maybe the name could be passed to each of (`Register`, `FromURL`, `FromString`) instead, so that the zero value of urlSchemeMap works?

Comment (vangent) [reply pubsub/pubsub.go]: We should also add a test for the zero value.

Comment (jba) [reply pubsub/pubsub.go]: Passing the name to each method.

`TestURLMux` does create a zero value. Did you mean something else?

**Comment by jba** [comment]
PTAL.

File: pubsub/pubsub.go (hunk 1 above)
Comment (vangent) [reply pubsub/pubsub.go]: I'm not sure how that test passed with your previous commit. It looks good now though.

Comment (vangent) [reply pubsub/pubsub.go]: Ah, I see, it was just missing the name; it always handled `m` being `nil`.
